### PR TITLE
TS Lua: ts.debug support debug flags may be better

### PR DESCRIPTION
**Pull request by scw00**

### CODE REVIEW

File: plugins/experimental/ts_lua/ts_lua_misc.c [hunk 1]
```diff
@@ -102,11 +102,25 @@ ts_lua_get_now_time(lua_State *L)
...
+
+  if (lua_gettop(L) == 2) {
+    msg   = luaL_checklstring(L, 2, &msg_len);
+    flag  = luaL_checklstring(L, 1, &flag_len);
+    int n = flag_len + 1 > MAX_FLAG_LEN - 1 ? MAX_FLAG_LEN - 1 : flag_len + 1;
+    memcpy(flags_buffer, flag, n);
+    flags_buffer[n] = 0;
+  } else {
+    msg = luaL_checklstring(L, 1, &msg_len);
+    memcpy(flags_buffer, TS_LUA_DEBUG_TAG, 6);
```
Comment (shukitchan) [inline plugins/experimental/ts_lua/ts_lua_misc.c]: Can we do away without memcpy() here and above?

File: doc/admin-guide/plugins/ts_lua.en.rst [hunk 2]
```diff
@@ -178,17 +178,18 @@ Here is an example:
 
 ts.debug
 --------
-**syntax:** *ts.debug(MESSAGE)*
+**syntax:** *ts.debug(TAG, MESSAGE)*
```
Comment (shukitchan) [inline doc/admin-guide/plugins/ts_lua.en.rst]: you are making this backward compatible in the code. Perhaps we can state that in the doc.

**Comment by scw00** [comment]
I'll update later!

File: plugins/experimental/ts_lua/ts_lua_misc.c (hunk 1 above)
Comment (scw00) [inline plugins/experimental/ts_lua/ts_lua_misc.c]: Ummmm! It seems that Lua doesn't promise the string alway with '\0' ending(Lua Document), so I try to copy it.
Any idea to fix it ?

Comment (shukitchan) [inline plugins/experimental/ts_lua/ts_lua_misc.c]: perhaps lua to the rescue? - https://www.lua.org/pil/27.2.html

**Comment by PSUdaemon** [comment]
[approve ci]

**Comment by shukitchan** [comment]
[approve ci]

**Comment by scw00** [comment]
That coredump could be fixed after amc or oknet finish the thread issue. Please run again!

**Comment by shukitchan** [comment]
yeah. otherwise it is fine.
sorry i forgot about this PR and only just remembered it.

**Comment by scw00** [comment]
That's all right. By the way, Happy Dragon Boat Festival!

**Comment by shukitchan** [comment]
[approve ci]

**Pull request merged**